ui/views.py

```diff
@@ -334,6 +338,22 @@ class VideoViewSet(ModelDetailViewset):
         ui_permissions.HasVideoPermissions,
     )
 
+    @detail_route()
+    def analytics(self, request, key=None):
+        """get video analytics data"""
+        # pylint: disable=unused-argument
+        if 'throw' in request.GET:
+            return HttpResponse(status=500)
+        if 'mock' in request.GET:
```

Comment:
Instead of putting this here, might it be possible to mock `get_video_analytics` in `views_test.py`, to keep test code and app code separate? Something like:
```python
mocker.patch('ui.views.get_video_analytics', side_effect= generate_mock_video_analytics_data)
```
and place the `generate_mock_video_analytics_data` function in the same test file?

Reply:
I put this 'mock' conditional in here for doing integration testing w/ front-end code. (for testing from the client-end sans the GoogleAnalytics query in the back-end).

Perhaps I should remove this, or at least explain its purpose better?

Reply:
Okay, I assumed it was for unit-testing.  Since it has a purpose outside of that I'm fine with leaving it here.

Reply:
Sounds good, I'll leave it in and add a descriptive comment.